public/components/datasources/components/new/auth_details.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiFormRow, EuiFieldText, EuiFieldPassword } from '@elastic/eui';
+import { useState } from 'react';
+import React from 'react';
+import { AuthMethod } from '../../../../../common/constants/data_connections';
+
+interface AuthDetailProps {
+  currentAuthMethod: AuthMethod;
+  currentPassword: string;
+  currentUsername: string;
+  currentAccessKey?: string;
+  currentSecretKey?: string;
+  currentRegion?: string;
+  setRegionForRequest?: React.Dispatch<React.SetStateAction<string>>;
+  setAccessKeyForRequest?: React.Dispatch<React.SetStateAction<string>>;
+  setSecretKeyForRequest?: React.Dispatch<React.SetStateAction<string>>;
+  setPasswordForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setUsernameForRequest: React.Dispatch<React.SetStateAction<string>>;
```

Comment:
I know we do it quite often so I'm more just curious: is there any way to send updaters in props without having to write out all these dispatch types and keep track of everything?